### ISSUE TITLE
False Positive Syntax Error in f-string

### ISSUE THREAD

**Issue by astromancer** [pappasam/jedi-language-server#150]
Not sure if this is the right place to report this, but maybe you can point me in the right direction if not...

In vscode with JediLSP, the following snippet triggers a SyntaxError, even though the code evaluates just fine:
```python
x = '...'
f'\{{{x}\}}'
```
![JediFalseSyntaxError](https://user-images.githubusercontent.com/5623552/122437725-42f1fb00-cf9a-11eb-9cd3-b19012e2bf0b.png)

**Comment by krassowski** [comment]
What about `rf'\{{{x}\}}'`? I think that neither `\{ ` nor `\}` is a valid escape sequence. You might be using a version of Python where this is not a `SyntaxError` yet, while Jedi is configured to run on a future version of Python.

For reference see: https://docs.python.org/3/reference/lexical_analysis.html:

> Changed in version 3.6: Unrecognized escape sequences produce a DeprecationWarning. In a future Python version they will be a SyntaxWarning and eventually a SyntaxError.

But if this still happens with `rf'\{{{x}\}}'`, then this is likely an issue in Jedi.

**Comment by pappasam** [comment]
Yep, it still happens with `rf'\{{{x}\}}'`. Seems like a bug in Jedi itself. @astromancer you might want to raise this issue here: https://github.com/davidhalter/jedi/issues . Seems like an edge case with f-string handling.

**Comment by astromancer** [comment]
@pappasam Thanks for checking. Will report this at jedi.
